core-bundle/src/Image/Studio/Figure.php

```diff
@@ -0,0 +1,379 @@
+     */
+    private $options;
+
+    /**
+     * Create a figure container.
+     *
+     * All arguments but the main image result can also be set via a Closure
+     * that returns the value instead. Closures will only be evaluated on
+     * demand.
+     *
+     * @param ImageResult                                                                 $image          Main image
+     * @param MetaData|(\Closure(self):MetaData|null)|null                                $metaData       Meta data container
+     * @param array<string, string|null>|(\Closure(self):array<string, string|null>)|null $linkAttributes Link attributes
+     * @param LightBoxResult|(\Closure(self):LightBoxResult|null)|null                    $lightBox       Light box
+     * @param array<string, mixed>|(\Closure(self):array<string, mixed>)|null             $options        Template options
+     */
+    public function __construct(ImageResult $image, $metaData = null, $linkAttributes = null, $lightBox = null, $options = null)
+    {
+        $this->image = $image;
+        $this->metaData = $metaData;
+        $this->linkAttributes = $linkAttributes;
+        $this->lightBox = $lightBox;
+        $this->options = $options;
+    }
+
+    /**
+     * Return the image result of the main resource.
+     */
+    public function getImage(): ImageResult
+    {
+        return $this->image;
+    }
+
+    /**
+     * Return if a light box result can be obtained.
+     */
+    public function hasLightBox(): bool
+    {
+        $this->resolveIfClosure($this->lightBox);
+
+        return $this->lightBox instanceof LightBoxResult;
+    }
+
+    /**
+     * Return the light box result (if available).
+     */
+    public function getLightBox(): LightBoxResult
+    {
+        if (!$this->hasLightBox()) {
+            throw new \LogicException('This result container does not include a light box.');
+        }
+
+        // Safely return as Closure will be evaluated at this point
+        return $this->lightBox;
+    }
+
+    public function hasMetaData(): bool
+    {
+        $this->resolveIfClosure($this->metaData);
+
+        return $this->metaData instanceof MetaData;
+    }
+
+    /**
+     * Return the main resource's meta data.
+     */
+    public function getMetaData(): MetaData
+    {
+        if (!$this->hasMetaData()) {
+            throw new \LogicException('This result container does not include meta data.');
+        }
+
+        // Safely return as Closure will be evaluated at this point
+        return $this->metaData;
+    }
+
+    /**
+     * Return a key-value list of all link attributes. This excludes "href" by
+     * default.
+     */
+    public function getLinkAttributes(bool $includeHref = false): array
+    {
+        $this->resolveIfClosure($this->linkAttributes);
+
+        if (null === $this->linkAttributes) {
+            $this->linkAttributes = [];
+        }
+
+        // Generate href attribute
+        if (!\array_key_exists('href', $this->linkAttributes)) {
```

Comment:
```suggestion
        if ($includeHref && !\array_key_exists('href', $this->linkAttributes)) {
```

Because I think the `!empty($this->linkAttributes['href'])` on line 159 will evaluate the closure and if `$includeHref` is `false`, the link attribute will be filtered out anyway.

Reply:
I first had it the way you're suggesting it but removed it again. The reason is, that the `rel` attribute depends on the `href` attribute. This must be correct no matter if you're querying the `href` or any other link attribute. You can add it and run the tests to see the difference. :smile_cat: 

btw.: The link attributes closure is evaluated anyways, the lightbox + metadata ones only if the href isn't defined.